### ISSUE TITLE
give TrafficLoggingFactory a public API to reset its counter

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2757]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2757|
|Type|enhancement|
|Created|2007-07-28 12:35:32Z|

TrafficLoggingFactory keeps track of how many connections have been made through it and uses this counter in constructing filenames to perform the logging.  It would be useful, particularly for unit tests, if the counter could be reset to 0 via some public API.

In current trunk, ```twisted.test.test_policies.LoggingFactoryTestCase.testThingsGetLogged``` fails when run with ```trial -u``` since it doesn't account for the fact that the counter was incremented by a previous test.

<details><summary>Searchable metadata</summary>

```
trac-id__2757 2757
type__enhancement enhancement
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1185626132000000 1185626132000000
changetime__1186569334000000 1186569334000000
version__None None
owner__ 
cc__therve cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

The work is done in trafficlogging-counter-2757.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_5"></a>|@ralphm set owner to @therve|
|-|-|

Looks good. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set status to `closed`|
|-|-|

(In [20937]) Merge trafficlogging-counter-2757

Author: therve
Reviewer: ralphm
Close [#2757](https://github.com/twisted/twisted/issues/2757)

Add a resetCounter method to twisted.protocol.policies.TrafficLoggingFactory,
used to reset the counter of connections to 0. Also, use this API in the tests
so that they pass with the --until-failure flag of trial.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve commented|
|-|-|

(In [20942]) Revert r20937, wrong API defined.

Refs [#2757](https://github.com/twisted/twisted/issues/2757)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve set status to `reopened`|
|-|-|

Sorry we can do better.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve removed owner<br>@therve set status to `new`|
|-|-|

OK, I cleaned it up a bit in [20974]. That should be better.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_10"></a>|@ralphm set owner to @therve|
|-|-|

Maybe split out the test for `_counter` and and `resetCounter`. Otherwise seems to better, indeed.

I'd be ok with merging this.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_11"></a>|@therve commented|
|-|-|

(In [20991]) Add a specific test for resetCounter.

Refs [#2757](https://github.com/twisted/twisted/issues/2757).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_12"></a>|@therve removed owner|
|-|-|

Thanks, I added another test. Still need a little review for the new docstrings.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_13"></a>|@ralphm set owner to @therve|
|-|-|

Works for me.

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_14"></a>|@therve set status to `closed`|
|-|-|

(In [21000]) Merge trafficlogging-counter-2757

Author: therve
Reviewer: ralphm
Fixes [#2757](https://github.com/twisted/twisted/issues/2757)

Add a resetCounter method to twisted.protocol.policies.TrafficLoggingFactory,
used to reset the counter of connections to 0. Also, refactor
TrafficLoggingProtocol so that it gets its counter from the factory instead of
holding it as a class variable.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
